management-api-agent-common/src/main/java/com/datastax/mgmtapi/NodeOpsProvider.java

```diff
@@ -740,39 +737,46 @@ public void clearSnapshots(
   @Rpc(name = "repair")
   public String repair(
       @RpcParam(name = "keyspaceName") String keyspace,
-      @RpcParam(name = "tables") Optional<List<String>> tables,
+      @RpcParam(name = "tables") List<String> tables,
       @RpcParam(name = "full") Boolean full,
       @RpcParam(name = "notifications") boolean notifications,
-      @RpcParam(name = "repairParallelism") Optional<RepairParallelism> repairParallelism,
-      @RpcParam(name = "datacenters") Optional<Collection<String>> datacenters,
-      @RpcParam(name = "associatedTokens") Optional<List<RingRange>> associatedTokens,
-      @RpcParam(name = "repairThreadCount") Optional<Integer> repairThreadCount)
+      @RpcParam(name = "repairParallelism") String repairParallelism,
+      @RpcParam(name = "datacenters") List<String> datacenters,
+      @RpcParam(name = "associatedTokens") String ringRangeString,
+      @RpcParam(name = "repairThreadCount") int repairThreadCount)
       throws IOException {
     // At least one keyspace is required
     assert (keyspace != null);
     Map<String, String> repairSpec = new HashMap<>();
-    repairParallelism.map(rPar -> repairSpec.put(RepairOption.PARALLELISM_KEY, rPar.getName()));
+    // add tables/column families
+    if (tables != null && !tables.isEmpty()) {
+      repairSpec.put(RepairOption.COLUMNFAMILIES_KEY, String.join(",", tables));
+    }
+    // set incremental reapir
     repairSpec.put(RepairOption.INCREMENTAL_KEY, Boolean.toString(!full));
-    repairThreadCount.map(
-        tCount ->
-            repairSpec.put(
-                RepairOption.JOB_THREADS_KEY, Integer.toString(tCount == 0 ? 1 : tCount)));
+    // Parallelism should be set if it's requested OR if incremental repair is requested.
+    if (!full) {
+      // Incremental repair requested, make sure parallelism is correct
+      if (repairParallelism != null
+          && !RepairParallelism.PARALLEL.getName().equals(repairParallelism)) {
+        throw new IOException(
+            "Invalid repair combination. Incremental repair if Parallelism is not set");
+      }
+      // Incremental repair and parallelism should be set
+      repairSpec.put(RepairOption.PARALLELISM_KEY, RepairParallelism.PARALLEL.getName());
+    }
+    // thread count should be at least 1
+    repairSpec.put(
+        RepairOption.JOB_THREADS_KEY, Integer.toString(Integer.max(1, repairThreadCount)));
```

Comment:
Question: Do we need to set this explicitly? I thought leaving it unset was best if the original request didn't contain the key.

Reply:
I guess we could then. But the logic then becomes something like:
```java
if (repairThreadCount > 1) {
    repairSpec.put(RepairOption.JOB_THREADS_KEY, Integer.toString(repairThreadCount));
}
```
So it's only added if the request asked for more than 1 thread.

Reply:
I think that's the right way, yep.